tests/CommunityToolkit.Mvvm.SourceGenerators.UnitTests/Test_SourceGeneratorsDiagnostics.cs

```diff
@@ -1801,6 +1801,25 @@ public class TestAttribute : Attribute
         VerifyGeneratedDiagnostics<RelayCommandGenerator>(source, "MVVMTK0038");
     }
 
+    [TestMethod]
+    public async Task AsyncVoidReturningRelayCommandMethodAnalyzer()
+    {
+        string source = """
+            using System;
+            using CommunityToolkit.Mvvm.Input;
+
+            public partial class MyViewModel
+            {
+                [RelayCommand]
+                private async void {|MVVMTK0039:TestAsync|}()
+                {
+                }
+            }
+            """;
+
+        await VerifyAnalyzerDiagnosticsAndSuccessfulGeneration<AsyncVoidReturningRelayCommandMethodAnalyzer>(source, LanguageVersion.CSharp8);
+    }
```

Comment:
This test looks like a duplicate of the two other tests?

Reply:
Mmh not sure I'm following. What would this be a duplicate of? 🤔

Reply:
Oh, my first thought is that this test verifies the existence of the diagnostic, and the other tests verifies both the diagnostic and codefix, in which case this test is redundant.

But I see now this test also verifies successful code generation.

There is a bit of overlap though, but not really duplicate.